### PR TITLE
Please Fix the problem. Thank You

### PR DESCRIPTION
**Pull request by septyaman**
Some improvements to troubleshoot problems that occur. There are some among the very important to promptly update. May be useful, thank you.

fix error, minify the the bugs. See in function . Why return a value that clearly does not exist?

    {
        if ( $secret=null)  {
            die;
        }
	//  an return undefined ?
       $this->_secret=$secret;
    }

shoul be,

    {
        if ( $secret=null)  {
            die;
       }
	// or
       else { $this->_secret=$secret;}
    }

and other small fix objects... thanks

### CODE REVIEW

File: php/recaptchalib.php [hunk 1]
```diff
@@ -33,9 +34,10 @@
 /**
  * A ReCaptchaResponse is returned from checkAnswer().
  */
+ 
 class ReCaptchaResponse
 {
-    public $success;
+    public $success=false;
```
Comment (septyaman) [inline php/recaptchalib.php]: Preventing the connection failure occurs and produces unexpected

File: php/recaptchalib.php [hunk 2]
```diff
@@ -52,13 +54,14 @@ class ReCaptcha
      *
      * @param string $secret shared secret between site and ReCAPTCHA server.
      */
-    function ReCaptcha($secret)
+    function ReCaptcha($secret=null)
     {
-        if ($secret == null || $secret == "") {
+        if ( empty($secret) ) {
```
Comment (septyaman) [inline php/recaptchalib.php]: To be more effective and targeted

File: php/recaptchalib.php [hunk 3]
```diff
@@ -52,13 +54,14 @@ class ReCaptcha
...
             die("To use reCAPTCHA you must get an API key from <a href='"
                 . self::$_signupUrl . "'>" . self::$_signupUrl . "</a>");
         }
-        $this->_secret=$secret;
+		
+        else $this->_secret=$secret;
```
Comment (septyaman) [inline php/recaptchalib.php]: his solves the problem, just give the value of existing

File: php/recaptchalib.php [hunk 4]
```diff
@@ -123,18 +126,17 @@ public function verifyResponse($remoteIp, $response)
                 'response' => $response
             )
         );
+		
         $answers = json_decode($getResponse, true);
         $recaptchaResponse = new ReCaptchaResponse();
 
-        if (trim($answers ['success']) == true) {
+        if ( isset($answers[0]) && trim($answers[0] ) == true) {
```
Comment (septyaman) [inline php/recaptchalib.php]: Prevent failure of connections, and only if it is true

File: php/recaptchalib.php [hunk 5]
```diff
@@ -123,18 +126,17 @@ public function verifyResponse($remoteIp, $response)
...
             $recaptchaResponse->success = true;
         } else {
             $recaptchaResponse->success = false;
-            $recaptchaResponse->errorCodes = $answers [error-codes];
+            $recaptchaResponse->errorCodes = isset($answers[1]) ? $answers[1] : null;
```
Comment (septyaman) [inline php/recaptchalib.php]: Prevent failure of connections, and only if it has a value

**Comment by Jerremyah** [comment]
You rock bro!

**Comment by wolffe** [comment]
Will the author merge these changes into the trunk? It seems there is no activity since the initial commit.

**Comment by septyaman** [comment]
:sleeping:

**Comment by rowan-m** [comment]
I believe these issues should be resolved in the new v1.1 If not, there's a proper contribution process now so please feel free to submit again and I'll be watching the PRs.

**Pull request closed without merging**